rdfunit-junit/src/main/java/org/aksw/rdfunit/junit/RdfUnitJunitRunner.java

```diff
@@ -0,0 +1,199 @@
+                throw new InitializationError("Methods marked @InputModel must return com.hp.hpl.jena.rdf.model.Model");
+            }
+        }
+    }
+
+    private List<FrameworkMethod> getInputModelMethods() throws InitializationError {
+        List<FrameworkMethod> inputModelMethods = getTestClass().getAnnotatedMethods(InputModel.class);
+        if (inputModelMethods.isEmpty()) {
+            throw new InitializationError("At least one method with @InputModel annotation is required!");
+        }
+        return inputModelMethods;
+    }
+
+    private void generateRdfUnitTestCases() throws InitializationError {
+        final SchemaSource schemaSourceFromOntology = createSchemaSourceFromOntology();
+        final Collection<TestCase> testCases = createTestCases();
+        for (Map.Entry<FrameworkMethod, Model> e :
+                getTestMethodsToInputModelsMergedWithControlledVocabulary().entrySet()) {
+            for (TestCase t : testCases) {
+                this.testCases.add(new RdfUnitJunitTestCase(t, schemaSourceFromOntology, e.getValue(), e.getKey()));
+            }
+        }
+    }
+
+    private Map<FrameworkMethod, Model> getTestMethodsToInputModelsMergedWithControlledVocabulary() throws
+            InitializationError {
+        if (controlledVocabulary == null) {
+            return getInputModels();
+        }
+        final Map<FrameworkMethod, Model> testMethodsToMergedModels = new LinkedHashMap<>();
+        for (Map.Entry<FrameworkMethod, Model> e : getInputModels().entrySet()) {
+            testMethodsToMergedModels.put(
+                    e.getKey(),
+                    ModelFactory.createDefaultModel().add(e.getValue()).add(controlledVocabulary)
+            );
+        }
+        return testMethodsToMergedModels;
+    }
+
+    private void createOntologyModel() {
+        ontologyModel = ModelFactory.createDefaultModel().read(getOntology().uri());
+    }
+
+    private synchronized Object getTestCaseInstance() throws InitializationError {
+        if (testCaseInstance == null) {
+            try {
+                testCaseInstance = getTestClass().getJavaClass().newInstance();
+            } catch (InstantiationException | IllegalAccessException e) {
+                throw new InitializationError(e);
+            }
+        }
+        return testCaseInstance;
+    }
+
+    private Map<FrameworkMethod, Model> getInputModels() throws InitializationError {
+        final Map<FrameworkMethod, Model> inputModels = new LinkedHashMap<>();
+        for (FrameworkMethod m : getInputModelMethods()) {
+            try {
+                final Model inputModel = (Model) m.getMethod().invoke(getTestCaseInstance());
+                if (inputModel == null) {
+                    throw new InitializationError(
+                            String.format("@InputModel: %s returned null!", m.getMethod().getName())
+                    );
+                }
+                inputModels.put(m, inputModel);
```

Comment:
Creating a new Model from the Input-Model and merging the probably existing CV-Model would avoid the entire `getTestMethodsToInputModelsMergedWithControlledVocabulary` Method.